moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -514,8 +513,10 @@ bool JointModelGroup::getEndEffectorTips(std::vector<const LinkModel*>& tips) co
       ROS_ERROR_NAMED("robot_model.jmg", "Unable to find end effector link for eef");
       return false;
     }
-
-    tips.push_back(eef_link);
+    // insert eef_link into tips, if it's not yet there (only consider unique tips)
+    auto insert_it = std::lower_bound(tips.begin(), tips.end(), eef_link);
+    if (insert_it == tips.end() || eef_link != *insert_it)
```

Comment:
It took me a second to figure out why you were calling `tips.insert()` here - at first glance it looks like you're inserting things at particular places in the vector. Actually though, you're always appending. Could you write `tips.push_back(eef_link)` instead? Also, this iterator can be `const auto` instead of `auto`, right?

Reply:
I need to use insert, because I want to keep a sorted vector (which allows usage of std::lower_bound to check for duplicates during insertion).